crates/router/src/db/dispute.rs

```diff
@@ -131,54 +135,739 @@ impl DisputeInterface for Store {
+
+            let _ = mockdb
+                .insert_dispute(create_dispute_new(DisputeNewIds {
+                    dispute_id: "dispute_2".into(),
+                    attempt_id: "attempt_1".into(),
+                    merchant_id: "merchant_2".into(),
+                    payment_id: "payment_1".into(),
+                    connector_dispute_id: "connector_dispute_1".into(),
+                }))
+                .await
+                .unwrap();
+
+            let found_disputes = mockdb
+                .find_disputes_by_merchant_id_payment_id("merchant_1", "payment_1")
+                .await
+                .unwrap();
+
+            assert_eq!(1, found_disputes.len());
+
+            disputes_eq(created_dispute, found_disputes.get(0).unwrap().clone());
+        }
+
+        mod update_dispute {
+            use masking::Secret;
+            use serde_json::Value;
+            use storage_models::{
+                dispute::DisputeUpdate,
+                enums::{DisputeStage, DisputeStatus},
+            };
+            use time::macros::datetime;
+
+            use crate::db::{
+                dispute::{
+                    tests::mockdb_dispute_interface::{create_dispute_new, DisputeNewIds},
+                    DisputeInterface,
+                },
+                MockDb,
+            };
+
+            #[allow(clippy::unwrap_used)]
+            #[tokio::test]
+            async fn test_update_dispute_update() {
+                let mockdb = MockDb::new(&Default::default()).await;
+
+                let created_dispute = mockdb
+                    .insert_dispute(create_dispute_new(DisputeNewIds {
+                        dispute_id: "dispute_1".into(),
+                        attempt_id: "attempt_1".into(),
+                        merchant_id: "merchant_1".into(),
+                        payment_id: "payment_1".into(),
+                        connector_dispute_id: "connector_dispute_1".into(),
+                    }))
+                    .await
+                    .unwrap();
+
+                let updated_dispute = mockdb
+                    .update_dispute(
+                        created_dispute.clone(),
+                        DisputeUpdate::Update {
+                            dispute_stage: DisputeStage::PreDispute,
+                            dispute_status: DisputeStatus::DisputeAccepted,
+                            connector_status: "updated_connector_status".into(),
+                            connector_reason: Some("updated_connector_reason".into()),
+                            connector_reason_code: Some("updated_connector_reason_code".into()),
+                            challenge_required_by: Some(datetime!(2019-01-10 0:00)),
+                            connector_updated_at: Some(datetime!(2019-01-11 0:00)),
+                        },
+                    )
+                    .await
+                    .unwrap();
+
+                assert_eq!(created_dispute.id, updated_dispute.id);
```

Comment:
Same here, feel free to derive `PartialEq` on these types instead of the manual asserts for each field.

Reply:
I'm a bit confuse, you want to remove the assert block with `asser_eq!(created_dispute, updated_dispute)` ?
But this will not validate witch field is updated or not, that's the right way to do the test ?

Reply:
You're right. Please ignore my comment.

Marking this resolved.